src/hotspot/share/gc/shenandoah/shenandoahPacer.cpp

```diff
@@ -249,32 +246,30 @@ void ShenandoahPacer::pace_for_alloc(size_t words) {
   JavaThread* current = JavaThread::current();
   if (current->is_attaching_via_jni() ||
       !current->is_active_Java_thread()) {
+    claim_for_alloc<true>(words);
     return;
   }
 
-  double start = os::elapsedTime();
-
-  size_t max_ms = ShenandoahPacingMaxDelay;
-  size_t total_ms = 0;
+  double const max_delay = static_cast<double>(ShenandoahPacingMaxDelay) / 1000;
+  double total_delay = 0;
 
-  while (true) {
+  double start = os::elapsedTime();
```

Comment:
While we are here, let's avoid some integer divisions and floating-point math. Try to rewrite this using `jlong os::elapsed_counter()`, which returns integer nanoseconds? Do the math in `jlong`-s.